python/torch_mlir/eager_mode/torch_mlir_dispatch.py

```diff
@@ -0,0 +1,272 @@
+    for i, arg in enumerate(schema.arguments):
+        kwarg = False
+        if arg.name in kwargs:
+            val = kwargs[arg.name]
+            kwarg = True
+        else:
+            val = args[i]
+
+        if is_tensor_type(arg.type) and val is not None:
+            inp = graph.addInput()
+            if isinstance(arg.type, torch.OptionalType):
+                inp.setType(arg.type.getElementType())
+            else:
+                inp.setType(arg.type)
+
+            if kwarg:
+                inp.setDebugName(arg.name)
+        else:
+            inp = graph.insertConstant(val)
+            inp.node().moveBefore(node)
+
+        node.addInput(inp)
+
+    if node.hasMultipleOutputs():
+        for outp in node.outputs():
+            graph.registerOutput(outp)
+    else:
+        graph.registerOutput(node.output())
+
+    fn_name = str(node).strip()
+    fn = torch._C._create_function_from_graph(fn_name, graph)
+    return fn
+
+
+def annotate_args_kwargs(
+    script_fun: torch._C.ScriptFunction,
+    normalized_args: List[Any],
+    normalized_kwargs: Dict[str, Any],
+):
+    unwrapped_normalized_args = tree_map(
+        lambda x: x.detach().contiguous().numpy() if isinstance(x, torch.Tensor) else x,
+        normalized_args,
+    )
+    unwrapped_normalized_kwargs = tree_map(
+        lambda x: x.detach().contiguous().numpy() if isinstance(x, torch.Tensor) else x,
+        normalized_kwargs,
+    )
+
+    annotations = []
+    tensor_args = []
+    for i, arg in enumerate(unwrapped_normalized_args):
+        if isinstance(arg, np.ndarray):
+            # TODO: remove once size zero dimensions are handled by torch-mlir.
+            shape = tuple(map(lambda x: x or -1, arg.shape))
+            annotations.append(
+                TorchTensorType(shape=shape, dtype=normalized_args[i].dtype)
+            )
+            tensor_args.append(arg)
+
+    # Pull out tensor kwargs and put them in positional order.
+    tensor_kwargs_flat = []
+    if unwrapped_normalized_kwargs:
+        tensor_kwargs = {}
+        arg_idxs = {
+            arg_name: i
+            for i, arg_name in enumerate(
+                [arg.name for arg in script_fun.schema.arguments]
+            )
+        }
+        for i, (kw, arg) in enumerate(unwrapped_normalized_kwargs.items()):
+            if isinstance(arg, np.ndarray):
+                tensor_kwargs[arg_idxs[kw]] = (arg, normalized_kwargs[kw].dtype)
+
+        for i in range(len(tensor_kwargs)):
+            arg, arg_dtype = tensor_kwargs[i]
+            annotations.append(TorchTensorType(shape=tuple(arg.shape), dtype=arg_dtype))
+            tensor_kwargs_flat.append(arg)
+
+    return annotations, tensor_args, tensor_kwargs_flat
+
+
+def write_back_to_mutable(
+    op_ods: JitOperator,
+    out: Union[np.ndarray, List[np.ndarray]],
+    all_tensor_args: List[np.ndarray],
+):
+    """
+    Because of how we pass values to the backend (by copying the tensor to a numpy array)
```

Comment:
There's actually two parts here. One is that we copy Torch to numpy, so we need a write-back. Secondly, Torch-MLIR right now only handles arguments with value semantics, so we need to manually fake it, which we can only do right now for these special cases.